src/ray/raylet/node_manager.cc

```diff
@@ -1027,55 +1090,63 @@ void NodeManager::SubmitTask(const Task &task, const Lineage &uncommitted_lineag
-          // is also dead.
-          TreatTaskAsFailed(spec);
+        // If this actor is alive, check whether this actor is local.
+        auto node_manager_id = actor_entry->second.GetNodeManagerId();
+        if (node_manager_id == gcs_client_->client_table().GetLocalClientId()) {
+          // If this actor is local, queue the task for local execution, bypassing
+          // placement.
+          EnqueuePlaceableTask(task);
         } else {
+          // The actor is remote. Forward the task to the node manager that owns
+          // the actor.
           // Attempt to forward the task. If this fails to forward the task,
           // the task will be resubmit locally.
           ForwardTaskOrResubmit(task, node_manager_id);
         }
       }
     } else {
-      // We do not have a registered location for the object, so either the
-      // actor has not yet been created or we missed the notification for the
-      // actor creation because this node joined the cluster after the actor
-      // was already created. Look up the actor's registered location in case
-      // we missed the creation notification.
-      // NOTE(swang): This codepath needs to be tested in a cluster setting.
-      auto lookup_callback = [this](gcs::AsyncGcsClient *client, const ActorID &actor_id,
-                                    const std::vector<ActorTableDataT> &data) {
-        if (!data.empty()) {
-          // The actor has been created.
-          HandleActorCreation(actor_id, data);
-        } else {
-          // The actor has not yet been created.
-          // TODO(swang): Set a timer for reconstructing the actor creation
-          // task.
-        }
-      };
-      RAY_CHECK_OK(gcs_client_->actor_table().Lookup(JobID::nil(), spec.ActorId(),
-                                                     lookup_callback));
+      if (!seen) {
+        // We do not have a registered location for the object, so either the
+        // actor has not yet been created or we missed the notification for the
+        // actor creation because this node joined the cluster after the actor
+        // was already created. Look up the actor's registered location in case
+        // we missed the creation notification.
+        // NOTE(swang): This codepath needs to be tested in a cluster setting.
+        auto lookup_callback = [this, spec](gcs::AsyncGcsClient *client,
+                                            const ActorID &actor_id,
+                                            const std::vector<ActorTableDataT> &data) {
+          if (!data.empty()) {
+            // The actor has been created.
+            if (actor_registry_.find(actor_id) == actor_registry_.end()) {
+              HandleActorNotification(actor_id, data);
+            }
+          } else {
+            // The actor has not yet been created. Try to reconstruct it.
+            reconstruction_policy_.ListenAndMaybeReconstruct(
+                spec.ActorCreationDummyObjectId());
```

Comment:
@stephanie-wang Adding `ListenAndMaybeReconstruct` caused an issue in a corner case. 
If we try reconstructing actor here, the reconstruction task may be submitted too fast, even before the task lease of the real actor creation task is acquired (actor creation task may be on another node).  In this case, the check in function `ResubmitTask` will fail (because the actor is reconstructed, even before it's created.

I think it might not be necessary to do reconstruction here, because if there's no entry in actor table, it's most likely that the actor creation task has finished yet, and it will eventually finish at some point later on.

What do you think?

Reply:
I see, that's actually a little strange because reconstruction won't get triggered until at least 200ms after `ListenAndMaybeReconstruct` gets called, which means that it's taking more than that to acquire the task lease for the actor creation task. Is there a test case where you saw this happening that I can help you debug?

I would be okay with adding an additional delay before trying reconstruction for actor creation tasks, but we must make sure to handle the case where a node dies before the actor first gets created.

Reply:
`test_actor_multiple_gpus_from_multiple_tasks` occasionally fails because of this.

Reply:
Adding additional delay can mitigate this issue. But unless the delay is super long, the above problem is always possible, right? E.g., when there's a temporary network disconnection. 
How long do you think makes sense?

Reply:
Hmm that test is actually kind of problematic, we've been looking into it in a separate PR (#3011) also. I did push a fix that may be related and another node failure test to check for node failure before initialization. If `test_actor_multiple_gpus_from_multiple_tasks` passes on Travis now, then I think it should be okay to not add the additional delay.

Reply:
Okay, so I actually updated the branch to skip resubmission of the actor creation task if its status is still `ALIVE`. So it will only be a fatal check if the status of the actor is `DEAD` and we try to recreate it. Let me know what you think.

Reply:
What will happen in the following case?

The original actor creation task hasn't been submitted. Then in the look-up callback, we'll try to reconstruct the actor and the actor is successfully "reconstructed". Then, the original actor creation task gets submitted and try to construct this actor. 

I didn't reproduce this case locally. But it seems that something will go wrong .

Reply:
Hmm I don't think this can happen since you can't submit a method on an actor unless it's been created, which means that the creation task must have been submitted.

It is true that reconstruction suppression could still fail based on timing issues, but in that case I think the second attempt to create the actor should just fail and clean up any leftover state. I prefer to keep the fatal check (for when the `AppendAt` of an `ALIVE` actor entry falis) for now, though, and we can revisit that case if it comes up.